### PR TITLE
VPN-6352 - Add linter for language names in extras.xliff

### PR DESCRIPTION
**Pull request by brizental**

### CODE REVIEW

File: scripts/ci/check_lang_names.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    root = tree.getroot()
+
+    found_languages = []
+    for trans_unit in root.findall('.//{urn:oasis:names:tc:xliff:document:1.2}trans-unit'):
+        unit_id = trans_unit.get('id')
+        if unit_id.startswith('languages.'):
+            found_languages.append(unit_id.split('.')[1])
+
+    ###
+    # 3. Check if the lists are different
```
Comment (brizental) [inline scripts/ci/check_lang_names.py:55]: nit: Be more precise
```suggestion
    # 3. Check if there are languages in the i18n folder not in the extras.xliff
```

File: scripts/ci/check_lang_names.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+if __name__ == "__main__":
+    script_path = os.path.dirname(os.path.abspath(__file__))
+
+    ###
+    # 1. Grab the list of expected languages in the i18n submodule
+
+    i18n_path = os.path.join(script_path, '..', '..', '3rdparty', 'i18n')
+    if not os.path.exists(i18n_path):
+        print(f"i18n submodule not found in path {i18n_path}")
+        sys.exit(1)
```
Comment (flodolo) [inline scripts/ci/check_lang_names.py:20]: You can [exit directly with a message](https://docs.python.org/3/library/sys.html#sys.exit)

```
        sys.exit(f"i18n submodule not found in path {i18n_path}")
```

File: scripts/ci/check_lang_names.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import xml.etree.ElementTree as etree
+
+
+if __name__ == "__main__":
+    script_path = os.path.dirname(os.path.abspath(__file__))
+
+    ###
+    # 1. Grab the list of expected languages in the i18n submodule
+
+    i18n_path = os.path.join(script_path, '..', '..', '3rdparty', 'i18n')
```
Comment (flodolo) [inline scripts/ci/check_lang_names.py:17]:
```suggestion
    i18n_path = os.path.join(
        script_path, os.path.pardir, os.path.pardir, "3rdparty", "i18n"
    )    
```

File: scripts/ci/check_lang_names.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    if not os.path.exists(i18n_path):
+        print(f"i18n submodule not found in path {i18n_path}")
+        sys.exit(1)
+
+    # The i18n folder has one folder per language
+    # and the name of the folder is the code of the language.
+    #
+    # Therefore the list of expected languages
+    # is the list of subfolder in that directory.
+    expected_languages = []
```
Comment (flodolo) [inline scripts/ci/check_lang_names.py:27]: List comprehension should work here

```
    expected_languages = [
        f for f in os.listdir(i18n_path)
        if os.path.isdir(os.path.join(i18n_path, f)) and not f.startswith(".")
    ]
```

File: scripts/ci/check_lang_names.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    found_languages = []
+    for trans_unit in root.findall('.//{urn:oasis:names:tc:xliff:document:1.2}trans-unit'):
+        unit_id = trans_unit.get('id')
+        if unit_id.startswith('languages.'):
+            found_languages.append(unit_id.split('.')[1])
+
+    ###
+    # 3. Check if there are languages in the i18n folder not in the extras.xliff
+
+    diff = [lang_code for lang_code in expected_languages if lang_code not in found_languages]
```
Comment (flodolo) [inline scripts/ci/check_lang_names.py:57]: Alternative

```
diff = list(set(expected_languages) - set(found_languages))
```

**Review by flodolo — APPROVED** [review]
Mostly Python nits, but overall makes sense.

File: .github/workflows/linters.yaml [hunk 6]
```diff
@@ -61,6 +61,10 @@ jobs:
         run: |
           scripts/utils/spellcheck.sh list
 
+      - name: Check for missing language names is extras.xliff
```
Comment (brizental) [inline .github/workflows/linters.yaml:64]:
```suggestion
      - name: Check for missing language names in extras.xliff
```

**Pull request merged**